### PR TITLE
REM 7 Enable/Disable api methods

### PR DESCRIPTION
**Pull request by anatolii15**

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -76,6 +76,8 @@ services:
       - user_keys:/root/.sawtooth/keys
     ports:
       - '${REMME_REST_API_PORT:-8080}:8080'
+    environment:
```
Comment (eugene-babichenko) [inline docker-compose.dev.yml]: This should be included with `env_file` directive.

File: remme/rest_api/__main__.py [hunk 2]
```diff
@@ -16,13 +16,13 @@
 import argparse
 from pkg_resources import resource_filename
 import connexion
-from connexion.resolver import RestyResolver
+from remme.rest_api.api_methods_validator import CustomResolver
```
Comment (eugene-babichenko) [inline remme/rest_api/__main__.py]: Maybe a more descriptive name is needed here?

File: remme/rest_api/api_methods_validator.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def access_denied_function(*args, **kwargs):
+    return {'error': 'This API method is disabled by node admin'}, 403
+
+
+class CustomResolver(RestyResolver):
+
+    def __init__(self, default_module_name, collection_endpoint_name='search'):
+        rules = os.environ['REMME_REST_API_AVAILABLE_METHODS']
```
Comment (eugene-babichenko) [inline remme/rest_api/api_methods_validator.py]: Consider using [`os.getenv`](https://docs.python.org/3/library/os.html#os.getenv).

File: remme/rest_api/api_methods_switcher.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        return super().resolve(operation)
+
+    def is_allowed_operation(self, operation):
+        request_path = get_clear_method_path(operation.path)
+        request_method = operation.method.upper()
+
+        if request_path not in self.allowed_operations:
+            return False
+
+        return '*' in self.allowed_operations[request_path] \
```
Comment (bogomazov) [inline remme/rest_api/api_methods_switcher.py]: can both return statements be united into one?

File: remme/rest_api/api_methods_switcher.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        if not self.allow_all_methods:
+            try:
+                self.allowed_operations = get_allowed_operations(rules)
+            except IndexError:
+                raise ValueError('Could not parse {} env var value'.format(REST_METHODS_ENV_KEY))
+
+        super().__init__(default_module_name, collection_endpoint_name)
+
+    def resolve(self, operation):
+        if not self.allow_all_methods and not self.is_allowed_operation(operation):
```
Comment (bogomazov) [inline remme/rest_api/api_methods_switcher.py]: would you like to get rid of "not" to ease the reading process?

Comment (bogomazov) [reply remme/rest_api/api_methods_switcher.py]:
```
if self.allow_all_methods or self.is_allowed_operation(operation):
      return super().resolve(operation)
return Resolution(access_denied_function, operation.path + operation.method)
```

Comment (bogomazov) [reply remme/rest_api/api_methods_switcher.py]: Place `allow_all_methods` into` is_allowed_operation`

**Pull request merged**